pkg/services/sqlstore/permissions/dashboard.go

```diff
@@ -32,17 +32,15 @@ func (d DashboardPermissionFilter) Where() (string, []interface{}) {
 			SELECT distinct DashboardId from (
 				SELECT d.id AS DashboardId
 					FROM dashboard AS d
-					LEFT JOIN dashboard AS folder on folder.id = d.folder_id
 					LEFT JOIN dashboard_acl AS da ON
 						da.dashboard_id = d.id OR
 						da.dashboard_id = d.folder_id
-					LEFT JOIN team_member as ugm on ugm.team_id = da.team_id
 					WHERE
 						d.org_id = ? AND
 						da.permission >= ? AND
 						(
 							da.user_id = ? OR
-							ugm.user_id = ? OR
+							da.team_id IN (SELECT team_id from team_member AS tm WHERE tm.user_id = ?) OR
```

Comment:
it looks like we could change to JOIN above since the WHERE condition will only be true if there is a row in dashboard_acl

Reply:
I think the idea here is that the SQL scheduler is better at optimizing this select rather than the JOIN 🤔

Reply:
I'm going to test it again undoing this change to check if it isn't a big issue for MySQL.

Reply:
Using the JOIN, MySQL takes more than 3 minutes to return the results. Using the subquery takes less ~30 - ~40 seconds.

Reply:
@sakjur Maybe MySQL loads in memory the whole table before do any search and using a subquery is able to get a subset of information.

Reply:
:+1: The subquery seems better then

Reply:
Maybe I wasn't clear here, I was referring to the LEFT JOIN on dashboard_acl above, which could be a JOIN instead because we refer to it in the WHERE clause.  It may or may not be more efficient that way.